### PR TITLE
i102 Add cancel button to create game screen

### LINKED ISSUE #102: Add cancel button to create game screen
#### Carried over from #23 
- [x] add a cancel button in appbar that: 
  - [x] deletes newGameViewModel data in gameservice
  - [x] pops back to lobby on tap 

create a game - 2

#### Clean up leftovers? 
- [x] change title to "Create game 1/2" 
- [ ] make "eye" icon button switch between hide password text and show password text (and switch icon) 
- [x] ? change all uses of "new game" to "create game"

### PR DESCRIPTION
**Pull request by nickmeinhold**
Fixes #102

### CODE REVIEW

**Comment by nickmeinhold** [comment]
@tamari-gray  I think it's cooked!  🧑‍🍳 

Very happy to split it up if it's an annoying review. Cheers mate!

File: lib/screens/create_game1/create_game_screen1.dart [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/material.dart';
```
Comment (nickmeinhold) [inline lib/screens/create_game1/create_game_screen1.dart:1]: this was just a rename, no new code

Comment (nickmeinhold) [reply lib/screens/create_game1/create_game_screen1.dart:1]: oh sorry, that's a lie!  it's a rename *and* new code... I suck

File: lib/screens/create_game1/create_game_screen1.dart [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  final _formKey = GlobalKey<FormState>();
+  bool _autoValidateForm = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      resizeToAvoidBottomInset: true,
+      appBar: AppBar(
+        automaticallyImplyLeading: false,
+        centerTitle: false,
```
Comment (nickmeinhold) [inline lib/screens/create_game1/create_game_screen1.dart:26]: new code

File: lib/screens/create_game1/create_game_screen1.dart [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        title: const Text('Create Game 1/2',
+            style: TextStyle(color: Colors.white)),
+        actions: [
+          FlatButton.icon(
```
Comment (nickmeinhold) [inline lib/screens/create_game1/create_game_screen1.dart:30]: new code

File: lib/screens/create_game1/create_game_screen1.dart (hunk 1 above)
Comment (nickmeinhold) [reply lib/screens/create_game1/create_game_screen1.dart:1]: I made a note of the new stuff I could find, it was pretty minor, just the UI changes mentioned in the issue

File: lib/screens/create_game1/create_game_screen1.dart (hunk 3 above)
Comment (tamari-gray) [reply lib/screens/create_game1/create_game_screen1.dart:30]: Love this style of pointing out where to look.  We're gonna be breezing through pr's soon!

File: lib/screens/create_game1/create_game_screen1.dart (hunk 2 above)
Comment (tamari-gray) [reply lib/screens/create_game1/create_game_screen1.dart:26]: sorry just clarifying @nickmeinhold, is new code everything below this comment?

File: test/unit_tests/game_service/game_service_test.dart [hunk 4]
```diff
@@ -20,18 +20,18 @@ void main() {
     expect(_gameService.currentGame.id, _secondSelectedGame.id);
   });
 
-  test('add name + password when submit valid data in newgamescreen1', () {
+  test('add name + password when submit valid data in creategamescreen1', () {
```
Comment (tamari-gray) [inline test/unit_tests/game_service/game_service_test.dart:23]: love it, thanks for testing @nickmeinhold !!

File: test/create_game_section/new_game1/new_game1_widget_test.dart [hunk 5]
```diff
@@ -2,13 +2,13 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
```
Comment (tamari-gray) [inline test/create_game_section/new_game1/new_game1_widget_test.dart:2]: so close!! last file name update snaeaked through (new_game1 => create_game1) 🐍

**Review by tamari-gray — CHANGES_REQUESTED** [review]

File: test/create_game_section/new_game1/new_game1_widget_test.dart (hunk 5 above)
Comment (nickmeinhold) [reply test/create_game_section/new_game1/new_game1_widget_test.dart:2]: ha! snaekey file name! OK I'm on it!  thanks mate 😄

**Review by tamari-gray — APPROVED** [review]
:beers: thanks for the good work nick!

**Pull request merged**